### PR TITLE
[*] Records List - Improve performance when filtering on belongs to

### PR DESCRIPTION
**Pull request by VincentMolinie**
# REQUIRES
- https://github.com/ForestAdmin/forest-express/pull/274

## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by VincentMolinie** [comment]
[P1 - Mongoose Liana - Performance issues on collections having references (after v4 upgrade)](https://trello.com/c/lbUJKeJs/3833-p1-mongoose-liana-performance-issues-on-collections-having-references-after-v4-upgrade)

**Comment by VincentMolinie** [comment]
The test are failing in the CI because it requires the new forest-express (Works perfectly fine on local)

File: src/services/filters-parser.js [hunk 1]
```diff
@@ -13,16 +13,17 @@ function FiltersParser(model, timezone, options) {
...
-  this.formatAggregation = (aggregator, formatedConditions) => {
+  this.formatAggregation = async (aggregator, formatedConditions) => {
     const aggregatorOperator = this.formatAggregatorOperator(aggregator);
     return { [aggregatorOperator]: formatedConditions };
   };
 
-  this.formatCondition = (condition) => {
+  this.formatCondition = async (condition) => {
     if (_.isEmpty(condition)) {
+      console.log('Condition', condition);
```
Comment (VincentMolinie) [inline src/services/filters-parser.js]: To remove

File: src/services/filters-parser.js [hunk 2]
```diff
@@ -132,14 +133,85 @@ function FiltersParser(model, timezone, options) {
...
   this.formatField = field => field.replace(':', '.');
 
-  this.getAssociations = filtersString => BaseFiltersParser.getAssociations(filtersString);
+  this.getAssociations = async filtersString => BaseFiltersParser.getAssociations(filtersString);
+
+  this.formatAggregationForReferences = (aggregator, conditions) => ({ aggregator, conditions });
+
+  this.formatConditionForReferences = async (condition) => {
+    if (_.isEmpty(condition)) {
+      console.log('Condition 2', condition);
```
Comment (VincentMolinie) [inline src/services/filters-parser.js]: To remove

File: test/tests/services/resources-getter.js [hunk 3]
```diff
@@ -24,12 +24,12 @@ describe('Service > ResourcesGetter', () => {
           isCompositePrimary: false,
           searchFields: ['amount', 'comment'],
           fields: [
-            { field: '_id', type: 'ObjectId' },
+            { field: '_id', type: 'String' },
```
Comment (rap2hpoutre) [inline test/tests/services/resources-getter.js]: Why is it a `String` now? Does this have an impact?

Comment (VincentMolinie) [reply test/tests/services/resources-getter.js]: Because ObjectId was a mistake. I don't know who did this but ObjectId never existed in our code base

**Pull request merged**